main.tf

```diff
@@ -0,0 +1,202 @@
+      name         = backend_address_pool.value.name
+      ip_addresses = var.destinations
+    }
+  }
+
+  dynamic "probe" {
+    for_each = [for app in var.frontends : {
+      name = app.name
+      host = app.custom_domain
+      path = lookup(app, "health_path_override", "/health/liveness")
+    }]
+
+    content {
+      interval            = 10
+      name                = probe.value.name
+      host                = probe.value.host
+      path                = probe.value.path
+      protocol            = "Http"
+      timeout             = 15
+      unhealthy_threshold = 3
+    }
+  }
+
+  dynamic "backend_http_settings" {
+    for_each = [for app in var.frontends : {
+      name                  = app.name
+      cookie_based_affinity = try(title(app.appgw_cookie_based_affinity), "Disabled")
+    }]
+
+    content {
+      name                  = backend_http_settings.value.name
+      probe_name            = backend_http_settings.value.name
+      cookie_based_affinity = backend_http_settings.value.cookie_based_affinity
+      port                  = 80
+      protocol              = "Http"
+      request_timeout       = 30
+    }
+  }
+
+  dynamic "http_listener" {
+    for_each = [for app in var.frontends : {
+      name          = app.name
+      custom_domain = app.custom_domain
+    }]
+
+    content {
+      name                           = http_listener.value.name
+      frontend_ip_configuration_name = "appGwPrivateFrontendIp"
+      frontend_port_name             = "http"
+      protocol                       = "Http"
+      host_name                      = http_listener.value.custom_domain
+    }
+  }
+
+  dynamic "request_routing_rule" {
+    for_each = [for app in var.frontends : {
+      name = app.name
+    }]
+
+    content {
+      name                       = request_routing_rule.value.name
+      rule_type                  = "Basic"
+      http_listener_name         = request_routing_rule.value.name
+      backend_address_pool_name  = request_routing_rule.value.name
+      backend_http_settings_name = request_routing_rule.value.name
+      rewrite_rule_set_name      = local.x_fwded_proto_ruleset
+    }
+  }
+
+  rewrite_rule_set {
+    name = local.x_fwded_proto_ruleset
+
+    rewrite_rule {
+      name          = local.x_fwded_proto_ruleset
+      rule_sequence = 100
+
+      request_header_configuration {
+        header_name  = "X-Forwarded-Proto"
+        header_value = "https"
+      }
+
+      request_header_configuration {
+        header_name  = "X-Forwarded-Port"
+        header_value = "443"
+      }
+
+      request_header_configuration {
+        header_name  = "X-Forwarded-For"
+        header_value = "{var_add_x_forwarded_for_proxy}"
+      }
+    }
+  }
+}
+
+data "azurerm_monitor_diagnostic_categories" "diagnostic_categories" {
+  count = length(var.frontends) != 0 ? 1 : 0
+
+  resource_id = azurerm_application_gateway.ag[0].id
+}
+
+data "azurerm_log_analytics_workspace" "log_analytics" {
```

Comment:
this should be a workspace ID and not looking up the workspace directly

Reply:
Will modify the root and child module accordingly